docs/buffering.md

```diff
@@ -0,0 +1,74 @@
+### Producer SDK buffering
+
```

Comment:
One thing that might be useful to include is why we require temporal view of the buffer before diving into the specifics.